package.json

```diff
@@ -102,5 +102,11 @@
     "vite": "^4.2.1",
     "vite-plugin-css-injected-by-js": "^3.1.0",
     "webpack": "^5.75.0"
+  },
+  "dependencies": {
+    "@hookform/resolvers": "^3.3.2",
+    "react-hook-form": "^7.47.0",
+    "react-responsive-modal": "^6.4.2",
+    "zod": "^3.22.4"
```

Comment:
Sharing my comments about adding these dependencies here as well. We usually try to keep the dependencies few as possible to keep a small size for the bundled version of our library as well. I believe some dependencies might be convenient but some are unnecessary also and don't save us much dev time. I'm tagging @esezen here as well to see what he thinks. 

Sharing gzipped sizes here for easier lookup
"@hookform/resolvers": "923b",
"react-hook-form": "7.4k",
"react-responsive-modal": "3.8k",
"zod": "13.5"

React is 2.7k gzipped for comparison

Reply:
I'm still leaning on the side of keeping react-hook-forms but maybe remove zod and hookform resolvers 😛 react-hook-forms helps us with state management which would otherwise be a little more work to track and implement.

Reply:
I'm with @mocca102 on this one. With all of our libraries, we take dependencies very seriously and try to have as few of them as possible. Customers also value this highly and even with [client-js](https://github.com/Constructor-io/constructorio-client-javascript) which has only a handful (we are removing 1 dependency soon and 1 in the mid-term), we heard concerns from customers about size. Right now this PR is getting the bundle size above 100KB with a 30% increase in size. 

I believe `@hookform/resolvers` and `zod` are for validation. Considering the amount of validation we do and the complexity, I think we can get rid of these. What do you think? For `react-hook-form`, I'm still in the middle of reviewing the PR so I'm not entirely sure about how it's used. If the work needed to remove it and have our own implementation is around a day's work, I would vote to have our implementation as well. If you think it's more, let's move this to Slack or a to a meeting and discuss internally. There are concerns with dependencies other than size as well.

Reply:
Oh yeah for sure, I can remove the validator libraries 👍 agreed it makes no sense to blow up our bundle size if we're just using it for one validation (email). I'm on the fence with react-hook-form though since it is pretty helpful in structuring code and maintaining states. It's definitely something I can replace with our own within a day as well but it's nice to have that library manage form states for us 🙂

Reply:
Removed the validators and resolvers here 👍 2d8e9c5387b458dbf10eb8f13ff7e0cff41b364f

Reply:
For `react-hook-form`, I still think it's better to implement our own instead of bringing a library to handle a single input but let me tag @jjl014 to see what he thinks

Reply:
Ahh no worries, if you feel strongly about it, I can remove it as well 👍 it'll just be a few states to maintain on our end

Reply:
Hey Team, I just took a look and I agree with @mocca102 and @esezen here. Generally we try to keep our libraries as lean as possible as we've had complaints from customers before in regards to some of our other libraries. They can be pretty stingy when it comes to size and every byte counts unfortunately. 😓 

From what I can tell looking at the code, it looks like we're using the package for a small portion of the component which can be handled by simple `useState` hooks (but please correct me if I'm wrong). For a package that takes up such a large size of our library, it doesn't seem worth it and probably makes more sense to roll our own.

Reply:
Makes sense to me as well 👍 have switched to using our own states. Thanks for clarifying @jjl014 @esezen @mocca102 !